distributed/dashboard/components/scheduler.py

```diff
@@ -559,6 +559,140 @@ def update(self):
+            self.memory_figure.add_tools(hover)
+            self.memory_figure.yaxis[0].formatter = NumeralTickFormatter(format="0.0 b")
+            self.memory_figure.xgrid.visible = False
+
+            # Network
+            self.network_source = ColumnDataSource(
+                {
+                    "size": [],
+                    "bandwidth": [],
+                }
+            )
+
+            self.network_figure = figure(
+                title="Network Bandwidth -- Computing ...",
+                tools="",
+                toolbar_location="above",
+                x_range=["1 kiB", "10kiB", "100kiB", "1 MiB", "10 MiB", "50 MiB"],
+                **kwargs,
+            )
+
+            self.network_figure.vbar(
+                x="size", top="bandwidth", width=0.9, source=self.network_source
+            )
+            hover = HoverTool(
+                mode="vline", tooltips=[("Bandwidth", "@bandwidth{0.00 b}/s")]
+            )
+            self.network_figure.add_tools(hover)
+            self.network_figure.yaxis[0].formatter = NumeralTickFormatter(
+                format="0.0 b"
+            )
+            self.network_figure.xgrid.visible = False
+
+            self.root = row(
+                self.memory_figure,
+                self.disk_figure,
+                self.network_figure,
+            )
+
+            self.memory_data = {
+                "size": [],
+                "bandwidth": [],
+            }
+            self.disk_data = {
+                "size": [],
+                "bandwidth": [],
+            }
+            self.network_data = {
+                "size": [],
+                "bandwidth": [],
+            }
+
+            async def f():
+                result = await self.scheduler.benchmark_hardware()
+
+                for size, bandwidth in result["disk"].items():
+                    self.disk_data["size"].append(size)
+                    self.disk_data["bandwidth"].append(bandwidth)
+
+                for size, bandwidth in result["memory"].items():
+                    self.memory_data["size"].append(size)
+                    self.memory_data["bandwidth"].append(bandwidth)
+
+                for size, bandwidth in result["network"].items():
+                    self.network_data["size"].append(size)
+                    self.network_data["bandwidth"].append(bandwidth)
+
+            self.scheduler.loop.add_callback(f)
+
+    def update(self):
+        if any(self.disk_data.values()):
+            if self.memory_figure.title.text == "Memory Bandwidth":
+                return
+            else:
+                update(self.disk_source, self.disk_data)
+                update(self.memory_source, self.memory_data)
+                update(self.network_source, self.network_data)
+                self.memory_figure.title.text = "Memory Bandwidth"
+                self.disk_figure.title.text = "Disk Bandwidth"
+                self.network_figure.title.text = "Network Bandwidth"
+
+
```

Comment:
Could you add logic that invalidates and reruns the measures when the list of connected workers changes?

Reply:
I think that I wasn't clear with my intentions.  My goal is just to present a very rough understanding of these numbers.  I'm looking to see roughly where bandwidths are in the range of 50 MiB/s to 10 GiB/s.  If I'm off by a factor of two then I'm actually totally ok with that.

It seems like you have higher ambitions for this work than I do.  It seems like you're focused on making sure that this is precise and accurate.  It might help me to understand your reasoning here.  

Personally, I'd like to get something simple up and then be done, improving it in the future if we need to.  Making this very good isn't currently high on my priority list.  If there are reasons why it should be then I'm happy to change that list around, but I'd love to have a brief conversation on prioritization beforehand if possible.  

There are *lots* of ways in which this can give us mildly incorrect results.  You've brought up a few but let's be honest, this is a pretty informal benchmark.  I think that that's ok.  If it isn't then let's talk about that larger problem first.

Reply:
I now see that the measure is rebuilt on a manual browser refresh. I thought it would persist until the scheduler restart. Good enough for me.